Sources/Widgets/Representations/ArrowHandleRepresentation/index.js

```diff
@@ -168,6 +250,40 @@ function vtkArrowHandleRepresentation(publicAPI, model) {
     model.internalPolyData.modified();
     outData[0] = model.internalPolyData;
   };
+
+  publicAPI.requestData = (inData, outData) => {
+    const currShape = publicAPI
+      .getRepresentationStates(inData[0])[0]
+      .getShape();
+    if (shapeName !== currShape) {
+      if (firstShape === false) {
+        firstShape = true;
+      } else {
+        model.orientation = [1, 0, 0];
+        model.viewMatrix = mat4.create();
```

Comment:
do not reset orientation nor viewMatrix when changing shape

Reply:
the fact of not reseting orientation and viewMatric can create some annoying effects when the user changes handles. For instance cube or cone would inherit the former arrow orientation which is not what was originally planned in the behavior.js file (e.g function isHandleGlyph2D). What is the best solution here: leaving reset parameters as it is on this comment or generalize the faceCamera behavior to all glyphs?

Reply:
faceCamera does not depend on the shape.

Maybe it is more a matter of ignoring faceCamera when the shape does not make sense